### ISSUE TITLE
bug: "a bytes-like object is required, not 'str'" when executing lambda functions

### ISSUE THREAD

**Issue by abarbu** [localstack/localstack#5432]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Executing nodejs lambda functions results in the following error:

```

2022-02-06T08:26:11.660:DEBUG:localstack.services.awslambda.lambda_utils: Function already exist: func6
2022-02-06T08:26:17.769:DEBUG:localstack.utils.run: Executing command: cd /tmp/localstack/zipfile.4cf3444b; unzip -o -q /tmp/localstack/zipfile.4cf3444b/original_lambda_archive.zip
2022-02-06T08:26:26.023:INFO:localstack.services.awslambda.lambda_executors: Empty event body specified for invocation of Lambda "arn:aws:lambda:us-east-1:000000000000:function:func7"
2022-02-06T08:26:26.023:INFO:localstack.services.awslambda.lambda_executors: Running lambda: arn:aws:lambda:us-east-1:000000000000:function:func7
2022-02-06T08:26:26.023:DEBUG:localstack_ext.services.awslambda.lambda_launcher: Executing docker separate execution hook for function arn:aws:lambda:us-east-1:000000000000:function:func7
2022-02-06T08:26:26.056:DEBUG:localstack.utils.docker_utils: Getting the entrypoint for image: localstack/lambda:nodejs8.10
2022-02-06T08:26:26.058:DEBUG:localstack.utils.docker_utils: Creating container with attributes: {'mount_volumes': None, 'ports': <PortMappings: {}>, 'cap_add': 'NET_ADMIN', 'dns': '127.0.0.1', 'additional_flags': '', 'workdir': None, 'command': 'index.apiTestHandler', 'detach': False, 'entrypoint': '/tmp/ac9d7f2f.sh', 'env_vars': {'AWS_ACCESS_KEY_ID': 'test', 'AWS_SECRET_ACCESS_KEY': 'test', 'AWS_REGION': 'us-east-1', 'DOCKER_LAMBDA_USE_STDIN': '1', 'LOCALSTACK_HOSTNAME': '172.17.0.2', 'AWS_ENDPOINT_URL': 'http://172.17.0.2:4566', 'EDGE_PORT': '443', '_HANDLER': 'index.apiTestHandler', 'AWS_LAMBDA_FUNCTION_TIMEOUT': '30', 'AWS_LAMBDA_FUNCTION_NAME': 'func7', 'AWS_LAMBDA_FUNCTION_VERSION': '$LATEST', 'AWS_LAMBDA_FUNCTION_INVOKED_ARN': 'arn:aws:lambda:us-east-1:000000000000:function:func7', 'NODE_TLS_REJECT_UNAUTHORIZED': '0', 'LOCALSTACK_DEBUG': '1'}, 'image_name': 'localstack/lambda:nodejs8.10', 'interactive': True, 'name': None, 'network': 'bridge', 'remove': True, 'self': <localstack.utils.docker_utils.SdkDockerClient object at 0x7f79c5575880>, 'tty': False, 'user': 'root'}
2022-02-06T08:26:26.084:DEBUG:localstack.utils.docker_utils: Copying file /tmp/localstack/zipfile.4cf3444b/. into 1af89aa8bf6ccceb9329ffd13d1cab5435b1a389a7700fd1c9edcaf51e97d4fd:/var/task
2022-02-06T08:26:26.178:DEBUG:localstack.utils.docker_utils: Copying file /tmp/localstack/ac9d7f2f.sh into 1af89aa8bf6ccceb9329ffd13d1cab5435b1a389a7700fd1c9edcaf51e97d4fd:/tmp/ac9d7f2f.sh
2022-02-06T08:26:26.247:DEBUG:localstack.utils.docker_utils: Starting container 1af89aa8bf6ccceb9329ffd13d1cab5435b1a389a7700fd1c9edcaf51e97d4fd
2022-02-06T08:26:26.764:DEBUG:localstack.services.awslambda.lambda_executors: Lambda arn:aws:lambda:us-east-1:000000000000:function:func7 result / log output:

>
2022-02-06T08:26:26.787:INFO:localstack.services.awslambda.lambda_executors: writing log to file '/tmp/localstack/lambda_k5yrl10q.log'
2022-02-06T08:26:26.801:INFO:localstack.services.awslambda.lambda_api: Error executing Lambda function arn:aws:lambda:us-east-1:000000000000:function:func7: a bytes-like object is required, not 'str' Traceback (most recent call last):
File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 795, in run_lambda
result = LAMBDA_EXECUTOR.execute(
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 460, in execute
return do_execute()
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 450, in do_execute
return _run(func_arn=func_arn)
File "/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py", line 158, in wrapped
raise e
File "/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py", line 154, in wrapped
result = func(*args, **kwargs)
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 437, in _run
raise e
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 433, in _run
result = self._execute(lambda_function, inv_context)
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 756, in _execute
result = self.run_lambda_executor(lambda_function=lambda_function, inv_context=inv_context)
File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/awslambda/lambda_extended.py.enc", line 573, in run_lambda_executor
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 677, in run_lambda_executor
output.output_file()
File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 1655, in output_file
f.write(self._stderr)
File "/usr/local/lib/python3.8/tempfile.py", line 473, in func_wrapper
return func(*args, **kwargs)
TypeError: a bytes-like object is required, not 'str'
```

### How are you starting LocalStack?

With the `localstack` script

### Steps To Reproduce

```
aws lambda create-function --function-name func1 --runtime nodejs8.10 --role r1 --handler index.apiTestHandler --timeout 30 --zip-file fileb://apiTestHandler.zip --endpoint-url=http://localhost:4566
aws lambda invoke --function-name func1 /tmp/foo.tmp --endpoint-url=http://localhost:4566
```

Where index.js contains the js below and is zipped up with: zip -r apiTestHandler.zip index.js

```
const apiTestHandler = (payload, context, callback) => {
  return JSON.stringify({
    payload,
    context
  });
};
module.exports = {
  apiTestHandler
};
```

### Environment

```markdown
- OS: Ubuntu 18.04
- LocalStack: 0.14.0
```

**Comment by abarbu** [comment]
Upon further debugging this happens when docker is configured for nvidia. Removing the nvidia runtime from docker fixes the issue.

Specifically when you have the JSON below in `/etc/docker/daemon.json`. It seems like something related to how lambdas execute is too fragile to deal with nvidia.

```
{
    "runtimes": {
        "nvidia": {
            "path": "nvidia-container-runtime",
            "runtimeArgs": []
        }
    },
    "default-runtime": "nvidia"
}
```

**Comment by mohamedbanhawi** [comment]
I am experiencing the same issue without configuring docker for nvidia. Any updates on this ?

**Comment by SergioDevAlcalde** [comment]
Same problem

**Comment by lucaspiller** [comment]
I also get this issue with 0.14.0, if I downgrade to 0.13.3 it works fine.

**Comment by Darksoulsong** [comment]
I second that downgrading the image to 0.13.3 solved the issue for me.

**Comment by iiian** [comment]
Downgrading to 0.13.3 worked for me as well.

**Comment by whummer** [comment]
Thanks for reporting @abarbu @mohamedbanhawi  @SergioDevAlcalde @Darksoulsong @lucaspiller @iiian . This should be fixed in #5578 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists - we'll create a new tagged release shortly that you can pin to. Thanks!

**Comment by whummer** [comment]
Closing as fixed - let us know if the problem persists.
